net/net.go

```diff
@@ -234,12 +234,37 @@ func (n *net) CreateThread(_ context.Context, id thread.ID, opts ...core.NewThre
 	return n.getThreadWithAddrs(id)
 }
 
-func (n *net) ensureUnique(id thread.ID) error {
-	_, err := n.store.GetThread(id)
-	if err == nil {
-		return lstore.ErrThreadExists
+func (n *net) ensureUnique(id thread.ID, key crypto.Key) error {
+	// Check if thread exists.
+	thrd, err := n.store.GetThread(id)
+	if errors.Is(err, lstore.ErrThreadNotFound) {
+		return nil
 	}
-	if !errors.Is(err, lstore.ErrThreadNotFound) {
+	if err != nil {
+		return err
+	}
+	// Early out if no log key required.
+	if key == nil {
+		return nil
+	}
+	// Ensure we don't already have this log in our store.
+	if _, ok := key.(crypto.PrivKey); ok {
+		// Ensure we don't already have our 'own' log.
+		if thrd.GetOwnLog().PrivKey != nil {
+			return lstore.ErrThreadExists
+		}
+		return nil
+	}
+	pubKey, ok := key.(crypto.PubKey)
```

Comment:
If this is a public key, just make sure we don't already have a log for this key.